libs/dal/src/repositories/message-template/message-template.entity.ts

```diff
@@ -11,6 +11,9 @@ export class MessageTemplateEntity {
 
   _creatorId: string;
 
+  // TODO: Due a circular dependency I can't import LayoutId from Layout.
+  _layoutId: string | null;
```

Comment:
This line `export { IEmailBlock, ITemplateVariable } from '../message-template';` shouldn't be exported from the `libs/dal/src/repositories/layout/types.ts`. Just import it directly, because this way it creates a circular dependency.
ex. file `libs/dal/src/repositories/layout/layout.entity.ts` should import `IEmailBlock` from `../message-template` or another suggestion to move these types to some shared folder...

Reply:
I want to review the types and do some tidy up. Will do it in a different PR and I will address this problem. Mainly I was facing that `message-template` and `layout` are dependant on each other (because sharing properties and the relationship through `LayoutId`). Tidying up types should help for that. 🙂